libs/ledger-live-common/src/generated/bridge/js.ts

```diff
@@ -15,10 +15,19 @@ import solana from "../../families/solana/bridge/js";
 import stellar from "../../families/stellar/bridge/js";
 import tezos from "../../families/tezos/bridge/js";
 import tron from "../../families/tron/bridge/js";
+import { of } from "rxjs";
 import { makeLRUCache } from "../../cache";
 import network from "../../network";
-import { withDevice } from "../../hw/deviceAccess";
+import { withDevicePromise } from "../../hw/deviceAccess";
 import { createBridges as polkadotCreateBridges } from "@ledgerhq/coin-polkadot/bridge/js";
+import { SignerFactory as PolkadotSignerFactory } from "@ledgerhq/coin-polkadot/lib/signer";
+import * as polkadotSigner from "@ledgerhq/hw-app-polkadot";
+const signerFactory: PolkadotSignerFactory = async (deviceId: string) => {
+  return await withDevicePromise(deviceId, (transport) =>
+    of(new polkadotSigner.default(transport))
+  );
+};
```

Comment:
Seems weird to have some coin (Polkadot) specific code here when Polkadot coin is supposed to be an external package.
I'm sure we can either move this signer factory definition in the Polkadot or even coin framework lib, and / or generalize it with Typescript generics

Maybe something like this (pseudocode):

```ts
const signerFactoryTest: SignerFactory<T> = (withDevicePromise: Function, signer: T) => async (deviceId: string) => {
  return await withDevicePromise(deviceId, (transport) =>
    of(new signer.default(transport))
  );
};
```

Reply:
I agree with you.
But I will add a task so we can set more constraints on the generic :)